workflow/cron/cron_facade.go

```diff
@@ -42,15 +45,19 @@ func (f *cronFacade) Delete(key string) {
 	delete(f.entryIDs, key)
 }
 
-func (f *cronFacade) AddJob(key, schedule string, cwoc *cronWfOperationCtx) error {
+func (f *cronFacade) AddJob(key, schedule string, cwoc *cronWfOperationCtx) (ScheduledTimeFunc, error) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
 	entryID, err := f.cron.AddJob(schedule, cwoc)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	f.entryIDs[key] = entryID
-	return nil
+
+	// Return a function to return the last scheduled time
+	return func() time.Time {
```

Comment:
how come we need this func? the facade hides entry ID presumably?

Reply:
Yes it does, furthermore we want to pass the function to the cron workflow context and want it to be modular